### PR TITLE
Fixed System.Tests.DateTimeTests.TryFormat_MatchesToString test

### PR DESCRIPTION
**Pull request by AlexRadch**
Fixed System.Tests.DateTimeTests.TryFormat_MatchesToString test to test all cultures.

`DateTime.TryFormat` has a bug for Utf8 version https://github.com/dotnet/runtime/blob/0570583829c7656778c2b0e8f20570627d6ecde1/src/libraries/System.Private.CoreLib/src/System/DateTime.cs#L1809-L1811

I can't fix this method because Visual Studio crashes on my computer when I try to debug it. So this PR is only for the test.

Anyone can fix `DateTime.TryFormat` method bug to pass this test.

### CODE REVIEW

**Comment by tarekgh** [comment]
@stephentoub looks there is a bug in the line https://github.com/dotnet/runtime/blob/783402c51818892f40fb7aec55ec15f9d4660671/src/libraries/System.Private.CoreLib/src/System/Globalization/DateTimeFormat.cs#L323

It casts the character while the character can have non-ASCII value. I recall you fixed a similar issue before but I don't recall which other place we fixed it.

File: src/libraries/System.Runtime/tests/System/DateTimeTests.cs [hunk 1]
```diff
@@ -2735,48 +2735,54 @@ internal struct SYSTEMTIME
         public static void TryFormat_MatchesToString(string format)
         {
             DateTime dt = DateTime.UtcNow;
-            string expected = dt.ToString(format);
-
-            // UTF16
+            foreach (var culture in CultureInfo.GetCultures(CultureTypes.AllCultures))
```
Comment (stephentoub) [inline src/libraries/System.Runtime/tests/System/DateTimeTests.cs:2738]:
```suggestion
            foreach (CultureInfo culture in CultureInfo.GetCultures(CultureTypes.AllCultures))
```

**Comment by tarekgh** [comment]
All tests passed now :-)

**Review by tarekgh — APPROVED** [review]
Thanks @AlexRadch for submitting this PR and reproduced the problem.

**Pull request merged**